usaspending_api/etl/management/commands/update_file_c_linkages_in_delta.py

```diff
@@ -85,6 +85,36 @@ def handle(self, *args, **options):
         # Setup Logger
         logger = get_jvm_logger(self.spark, __name__)
 
+        # Before we lose history after a clone or drop table of our int.financial_accounts_by_awards table
+        #   we need to identify FABA records (file C submissions) that were deleted.
+        #   This step is important because we need to identify which awards need to be unlinked
+        #   from file C submissions that are now deleted.
+        identify_faba_deletions_query = """
+            CREATE OR REPLACE TEMPORARY VIEW identify_faba_deletions_query AS (
+                SELECT DISTINCT faba.award_id
+                    FROM int.financial_accounts_by_awards faba
+
+                    LEFT JOIN raw.financial_accounts_by_awards rfaba
+                    ON faba.financial_accounts_by_awards_id = rfaba.financial_accounts_by_awards_id
+
+                    WHERE rfaba.financial_accounts_by_awards_id IS NULL
+            );
+        """
+        # If this command is running for the first time in an environment or in our test
+        #   suites, int.financial_accounts_by_awards isn't guarenteed to exist.
+        int_faba_exists = self.spark._jsparkSession.catalog().tableExists("int.financial_accounts_by_awards")
+        if int_faba_exists:
+            self.spark.sql(identify_faba_deletions_query)
+            # Update Awards whose file C submissions are now deleted
+            delete_sql = """
+                MERGE INTO int.awards AS aw
+                USING identify_faba_deletions_query AS deletes
+                ON aw.id = deletes.award_id
+                WHEN MATCHED
+                    THEN UPDATE SET aw.update_date = NOW();
+            """
+            self.spark.sql(delete_sql)
```

Comment:
Where you able to come up with a way to avoid updating Awards twice?

Reply:
I'm not confident that having two merge statements will negatively impact performance. Overall, the same number of records are being merged whether the two merge statements are combined or not.

Reply:
I've one ahead and combined the merge statements like you requested.

Reply:
> I'm not confident that having two merge statements will negatively impact performance. Overall, the same number of records are being merged whether the two merge statements are combined or not.

The same number of records may be merged, but the number of parquet files being rewritten could change with with multiple merges. 

For example, say we have 100 awards evenly distributed in order across 10 parquet files (Awards 1-10 in File 1, etc). Say the first merge updates Awards 6 and 35. This will cause files 1 and 3 to be rewritten as files 11 and 12 (with 1 and 3 being invalidated). In the second merge, Awards 4 and 38 are updated. This would cause files 11 and 12 to be rewritten, resulting in a total of 4 file rewrites, where performing a single merge with all 4 updates would have resulted in 2 rewrites.

Reply:
As far as your implementation goes, one thing I'd want to double check is `identify_faba_deletions_query` actually contains records by the time `all_updated_awards` is created. One concern that I have is because `identify_faba_deletions_query` is a view on top of the `int` and `raw` schemas, when the `int` table is rewritten to reflect the current state of the `raw` table, the underlying query will no longer return any results. 

May have to materialize the results somehow by pulling out the IDs into memory before recreating the `int` table.

I'd like to see a test case covering this scenario. I have a pretty robust testing framework setup for this command. Should hopefully be easy to update it to include this case. Let me know if you'd like me to walk you through it

Reply:
Done